packages/model-generator/src/s3_string_object_fetcher.ts

```diff
@@ -14,15 +15,31 @@ export class S3StringObjectFetcher {
    */
   fetch = async (uri: string) => {
     const { bucket, key } = this.parseS3Uri(uri);
-    const getSchemaCommandResult = await this.s3Client.send(
-      new GetObjectCommand({ Bucket: bucket, Key: key })
-    );
-    const schema = await getSchemaCommandResult.Body?.transformToString();
-    if (!schema) {
-      // eslint-disable-next-line amplify-backend-rules/prefer-amplify-errors
-      throw new Error('Error on parsing output schema');
+    try {
+      const getSchemaCommandResult = await this.s3Client.send(
+        new GetObjectCommand({ Bucket: bucket, Key: key })
+      );
+      const schema = await getSchemaCommandResult.Body?.transformToString();
+      if (!schema) {
+        // eslint-disable-next-line amplify-backend-rules/prefer-amplify-errors
+        throw new Error('Error on parsing output schema');
+      }
+      return schema;
+    } catch (caught) {
+      if (caught instanceof NoSuchBucket) {
+        throw new AmplifyFault(
+          'NoSuchBucketFault',
+          {
+            message: `${caught.name} the bucket, ${bucket} does not exist. \n
```

Comment:
what does this print out? ${caught.name} specifically

Reply:
`${caught.name}` prints out "`NoSuchBucket`"